### DISCUSSION TITLE
Is it possible to make magic properties internal to class?

### DISCUSSION THREAD

**Discussion by bravik** [vimeo/psalm#6792] [Q&A]
Hi
Like in Eloquent Active Record I have some class:

```
/**
 * @property string $someProperty
 */
class MyActiveRecord {}
```
Is it possible to  mark such properties as `internal` or `readonly`?
It's possible for normal properties, but seems like not for magic ones..

I'd like to prohibit any access to class magic properties outside the class itself, or at least prohibit any direct writes..

**Comment by weirdan** [comment]
Would declaring it readonly (with `@property-read`) suffice?

**Reply by bravik** [reply]
Probably not, because `@property-read` would trigger an error if I write to the propperty internally in that class.
I just want to isolate this publicly available data properties inside this class and prohibit direct usage in other places.
